client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/User.java

```diff
@@ -24,38 +24,59 @@
 
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 
 
 /**
- * An authenticated user
+ * A user to be utilized with security APIs.
+ * Can be an existing authenticated user or it can be a new user to be enrolled to the native realm.
  */
 public final class User {
 
     private final String username;
-    private final Collection<String> roles;
+    private final Set<String> roles;
```

Comment:
if `User` is both part of a response as well as a request, we have to acknowledge that `roles` can change order (through a round trip serialize-deserialize) but the `User` is still the same (from the `equals` POV).